networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/forks/versions/GossipForkSubscriptionsPhase0.java

```diff
@@ -216,7 +218,16 @@ public void publishAttestation(final ValidateableAttestation attestation) {
 
   @Override
   public void publishBlock(final SignedBeaconBlock block) {
-    blockGossipManager.publishBlock(block);
+    if (spec.fork(spec.computeEpochAtSlot(block.getSlot())).equals(fork)) {
+      blockGossipManager.publishBlock(block);
+    } else {
+      LOG.debug(
+          "Not publishing block {} ({}) due to it being on a different fork {}, expected {}",
+          block.hashTreeRoot(),
+          block.getSlot(),
+          spec.fork(spec.computeEpochAtSlot(block.getSlot())).getCurrentVersion(),
+          fork.getCurrentVersion());
+    }
```

Comment:
I don't think we need this check because this is only called via `GossipForkManager` which selects the right topic to use for the fork at the block's slot.